v20.2/cluster-setup-troubleshooting.md

```diff
@@ -479,6 +479,30 @@ To view command commit latency:
+
+In production, lease transfer upon node failure can take longer than expected. In {{ page.version.version }}, this is observed in the following scenarios:
+
+- **The leaseholder node for the liveness range fails.** The liveness range is a system range that [stores the liveness record](architecture/replication-layer.html#epoch-based-leases-table-data) for each node on the cluster. If a node fails and is also the leaseholder for the liveness range, operations cannot proceed until the liveness range is transferred to a new leaseholder and the liveness record is made available to other nodes. This can cause momentary cluster unavailability.
+
+- **The leaseholder node for `system.users` fails.** The cluster accesses the `system.users` table when authenticating a user. A customer attempting to log in to the cluster will encounter a delay while waiting for the `system.users` table to find a new leaseholder. In **v21.2.0 and later**, there is no login delay because the authentication data is cached.
+
+- **A node comes in and out of liveness, due to network failures or overload.** A "flapping" node can repeatedly lose and acquire leases in between its intermittent heartbeats, causing wide-ranging unavailability. In **v21.1.6 and later**, a node must successfully heartbeat for 30 seconds after failing a heartbeat before it is eligible to acquire leases.
+
+- **Network or DNS issues cause connection issues between nodes.** If there is no live server for the IP address or DNS lookup, connection attempts to a node will not return an immediate error, but will hang until timing out. This can cause unavailability and prevent a speedy movement of leases and recovery. In **v20.2.18, v21.1.12, and later**, CockroachDB avoids contacting unresponsive nodes or DNS during certain performance-critical operations, and the connection issue should generally resolve in 10-30 seconds. However, an attempt to contact an unresponsive node could still occur in other scenarios that are not yet addressed.
+
+- **A node's disk stalls.** A disk stall on a node can cause write operations to stall indefinitely, causes the node's heartbeats to fail since the storage engine cannot write to disk as part of the heartbeat, and may cause read requests to fail if they are waiting for a conflicting write to complete. Lease acquisition from this node can stall indefinitely until the node is shut down or recovered. Pebble detects most stalls and will terminate the `cockroach` process after 60 seconds, but there are gaps in its detection. In **v21.2.13, v22.1.2, and later**, each lease acquisition attempt on an unresponsive node times out after 6 seconds. However, CockroachDB can still appear to stall as these timeouts are occurring.
```

Comment:
"A disk stall on a node can cause write operations to stall indefinitely, causes the node's heartbeats to fail" 

Should this be: "indefinitely, causing the node's..." OR "indefinitely, which causes the nodes'"